.travis.yml

```diff
@@ -1,21 +1,41 @@
-sudo: required
-dist: trusty
 language: node_js
-node_js: "6"
-env:
-  - CXX=g++-4.8
+node_js:
+  - 6
 cache: yarn
+
+# Define global C++ compiler version
+env:
+  global:
+    - CXX=g++-4.8
+  matrix:
+    - TEST_ENV=node
+    - TEST_ENV=browser
+before_install:
+  # Yarn defaults to an old version, make sure we
+  # get an up to date version
+  - npm install -g yarn
+before_script:
+  - export PATH=$PATH:./node_modules/.bin
+  - mkdir -p tests/config && echo "$PROJECT_CONFIG" > tests/config/project.json
+script:
+  gulp test --suite="$TEST_SUITE" --env="$TEST_ENV"
+
+# Integration Test suite
+jobs:
+  include:
+    - stage: Integration Tests
+      script: ./integration/webpack/runner.sh
```

Comment:
I've tried something to this affect, I believe doing this would run the tests in sequence rather than in parallel which, though functional, would be slow.